src/index.js

```diff
@@ -1,36 +1,147 @@
+      editId = i;
+      isEditedTask = true;
+      taskInput.value = tname;
+    };
+  }
+}
+editTask();
+
+function deleteTask() {
+  const delTas = document.querySelectorAll('.task-menu');
+  for (let k = 0; k < delTas.length; k += 1) {
+    // eslint-disable-next-line no-loop-func
+    delTas[k].lastElementChild.onclick = () => {
+      todos.splice(k, 1);
+      localStorage.setItem('todo-list', JSON.stringify(todos));
+      showTodo('all');
+    };
+  }
+}
+deleteTask();
+
+function addElliplistiners() {
+  const showMe = document.querySelectorAll('.uil.uil-ellipsis-h');
+  for (let x = 0; x < showMe.length; x += 1) {
+    showMe[x].onclick = () => {
+      const taskMenu = showMe[x].parentElement.lastElementChild;
+      taskMenu.classList.add('show');
+      document.addEventListener('click', (e) => {
+        // Removing show class from the task Menu on document click
+        if (e.target.tagName !== 'I' || e.target !== showMe[x]) {
+          taskMenu.classList.remove('show');
+        }
+      });
+      editTask();
+      deleteTask();
+    };
+  }
+}
+
+clearAll.addEventListener('click', () => {
+  // Removing all tasks from array/todos
+  todos.splice(0, todos.length);
+  localStorage.setItem('todo-list', JSON.stringify(todos));
+  showTodo('all');
+});
+
+// eslint-disable-next-line no-unused-vars
+const updateStatus = (selectedTask) => {
+  // getting paragraph that contains task name
+  const taskName = selectedTask.parentElement.lastElementChild;
+  if (selectedTask.checked) {
+    taskName.classList.add('checked');
+    todos[selectedTask.id].status = 'completed';
+  } else {
+    taskName.classList.remove('checked');
+    todos[selectedTask.id].status = 'pending';
+  }
+  localStorage.setItem('todo-list', JSON.stringify(todos));
+};
+
+taskInput.addEventListener('keyup', (e) => {
+  const userTask = taskInput.value.trim();
+  if (e.key === 'Enter' && userTask) {
+    if (!isEditedTask) {
+      // if edited task isn't true
+      if (!todos) {
+        // if todo-list doesn't exist, pass an empty array to todos
+        todos = [];
+      }
 
-const tasks = [
-  {
-    description: 'Cook',
-    completed: false,
-    index: 0,
-  },
-  {
-    description: 'Watch tv',
-    completed: false,
-    index: 1,
-  },
-  {
-    description: 'Complete Project',
-    completed: false,
-    index: 2,
-  },
-];
-
-tasks.sort((a, b) => a.index - b.index);
-
-tasks.forEach((task) => {
-  const list = document.createElement('li');
-
-  list.innerHTML = `
-          <label for="${task.index}">
-            <input type="checkbox" name="task" id="${task.index}">
-            <p>${task.description}</p>
-          </label>
-            <i class="fa-solid fa-ellipsis-vertical"></i> 
-              `;
-  taskBox.append(list);
+      const taskInfo = { name: userTask, status: 'pending' };
+      // adding a new task to todos
+      todos.push(taskInfo);
```

Comment:
- [ ] Here, where you create the to-do item, please ensure to be consistent with the keys. In this case, the `index` key is missing. The index is is important especially when you want to delete or update an item. 
- [ ] Also, kindly change the `status` key to `completed` and ensure that it is set to `false` for any new to-do item that is created 👍